### ISSUE TITLE
False positive for aria-required-children rule on role 'combobox'

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#548]
Aria 1.1 standard states child role ‘listbox’ is only needed when aria-expanded is set to true

Looks like Axe is following the Aria1.0 standard for role combobox . It is expecting both ‘listbox’ and ‘textbox’ as child roles, irrespective of the state of aria-expanded’.

![image](https://user-images.githubusercontent.com/22068434/30885509-6c01ea4e-a2c8-11e7-8660-4c5195248265.png)

**Comment by marcysutton** [comment]
We have to do some testing to find out the accessibility supported status of this, but we'll report back as soon as we have more information. Test cases welcome

**Comment by marcysutton** [comment]
@iamrafan if you have a working demo we can check out, that will make this go much faster.

**Comment by iamrafan** [comment]
@marcysutton : Please find the sample tests from below
```
case-1: Valid Case - aria-expanded is true and textbox, listbox are direct children of combobox
<div aria-label="Tag" role="combobox" aria-expanded="true" aria-haspopup="listbox">
    <input type="search" role="textbox" aria-label="input-1">
    <ul role="listbox">
        <li role="option">Zebra</li>
        <li role="option" id="selected_option">Zoom</li>
    </ul>
</div>

<hr> case-2: Valid case - aria-expanded is true, textbox is direct child of combobox and element with role listbox is owned by
combobox
<div aria-expanded="true" role="combobox" aria-owns="Basicdrop2-list">
    <span id="Basicdrop2-option" role="textbox" aria-readonly="true">
        <span>Select an Option</span>
    </span>
</div>

<ul role="listbox" id="Basicdrop2-list">
    <li role="option">Zebra</li>
    <li role="option" id="selected_option2">Zoom</li>
</ul>

<hr> case-3: False positive: aria-expanded is false and child textbox is present, no need of role listbox
<div aria-expanded="false" role="combobox">
    <input type="search" role="textbox" aria-label="input-3">
</div>

<br>
<hr> case-4: Valid: aria-expaned is true and role combobox is missing child role listbox
<div aria-expanded="true" role="combobox">
    <span id="Basicdrop3-option" role="textbox">
        <span>Select an Option</span>
    </span>
</div>
```

**Comment by isner** [comment]
I conducted some support testing. I was able to confirm that both Firefox/NVDA and Safari/VO combinations behave as expected even when listbox role is absent until the combobox is expanded. I'll put together a pull request with the appropriate changes.

**Comment by isner** [comment]
This issue is addressed in the following pull request: https://github.com/dequelabs/axe-core/pull/559

**Comment by iamrafan** [comment]
@isner , thank you.

**Comment by WilcoFiers** [comment]
PR merged, closing the issue.
